app/hooks/useCollectionTrees.ts

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import { NavigationNode, NavigationNodeType } from "@shared/types";
+import Collection from "~/models/Collection";
+import useStores from "~/hooks/useStores";
+
+export default function useCollectionsTree() {
```

Comment:
It would be good to explicitly define the return type here and add a jsdoc comment